pkg/kubeconfig/kubeconfig.go

```diff
@@ -25,15 +30,90 @@ func FromEnvironment() string {
 	return os.Getenv(EnvName)
 }
 
-type missingFileError struct {
-	Path string
+// Loader abstracts the unmarshaling of a kubeconfig file into a Config.
+//
+// This is essentially building an interface around the clientcmd.Load
+// function so that it can be swapped out for tests.
+type Loader interface {
```

Comment:
Crazy annoying idea, I wonder if we can just omit the concept of loader? I get why this is useful for testing, so we don't need to depend on the `client-go` implementation logic. But would it be that bad if we do?

It shouldn't be that difficult to pass a valid/invalid kubeconfig body. And I think it would simplify this code a lot.

Reply:
It seems that both you and Chris have had the same idea here, and it sounds like a good one to me. I'll whip something up.